library/src/main/java/kwasm/format/text/Utils.kt

```diff
@@ -121,11 +124,33 @@ fun CharSequence.parseStringChar(
 ): StringChar = codePoints().toArray().parseStringChar(index, inoutVal, context)
 
 @UseExperimental(ExperimentalUnsignedTypes::class)
+fun CharSequence.toUInt(context: ParseContext? = null): UInt {
+    if ("-" in this) {
+        throw ParseException("Negative values are not allowed in limits", context)
+    }
+    var powerVal = 1.toUInt()
+    var power = 0
+    val value = ByteArray(this.length)
+    repeat(this.length) { index -> value[index] = this.parseDigit(index, context) }
+    val convertedValue = value.foldRightIndexed(0.toUInt()) { _, charVal, acc ->
+        if (power > 0) {
+            powerVal *= 10u
+        }
+        power++
+        val byteVal = charVal.toInt()
+        acc + byteVal.toUInt() * powerVal
+    }
+    if (convertedValue.toString() != this) {
+        throw ParseException("Value overflow, specified value was larger than UInt.MAX_VALUE", context)
+    }
+    return convertedValue
+}
+
```

Comment:
You could use `Num` rather than needing to add a new parse method to utils?

Reply:
Or `IntegerLiteral.Unsigned`

Reply:
Done